### PR TITLE
[Compute] `az disk create`: `--gallery-image-reference` adds support for creating disk from shared gallery image version or community gallery image version

### LINKED ISSUE #22425: Support Put disk for Community Gallery and Shared Gallery
**Related command**

az disk create -g MyResourceGroup -n MyDisk --gallery-image-reference

**Resource Provider**

Az.Compute

**Description of Feature or Work Requested**

Currently users can create a disk from gallery image version by:
az disk create -g MyResourceGroup -n MyDisk --gallery-image-reference /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myRG/providers/Microsoft.Compute/galleries/myGallery/images/myImage/versions/1.0.0

and backend match the --gallery-image-reference input to [disk - properties - creationData - galleryImageReference - id]

for community gallery, user can do 
az disk create -g MyResourceGroup -n MyDisk --gallery-image-reference /CommunityGalleries/{communityGalleryPublicGalleryName}/Images/{imageName}/Versions/1.0.0

backend should match the --gallery-image-reference input to [disk - properties - creationData - galleryImageReference - communityGalleryImageId]

for shared gallery/direct sharing, user can do 
az disk create -g MyResourceGroup -n MyDisk --gallery-image-reference /SharedGalleries/{sharedGalleryUniqueName}/Images/{imageName}/Versions/1.0.0

backend should match the --gallery-image-reference input to [disk - properties - creationData - galleryImageReference - sharedGalleryImageId]

**Minimum API Version Required**

2022-03-02

**Swagger Link**

https://github.com/Azure/azure-rest-api-specs/pull/18945 (this has not been merged to main, but it should be soon)

**Target Date**

06/30

### PR DESCRIPTION
**Pull request by yanzhudd**
**Description**

Add two supported formats of `--gallery-image-reference` argument in `az disk create`scenario to support creating disk from shared gallery image version and community gallery image version.
Close #22425

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compute

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 1]
```diff
@@ -181,7 +181,8 @@ def load_arguments(self, _):
         c.argument('disk_mbps_read_only', type=int, help='The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly. MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10')
         c.argument('image_reference', help='ID or URN (publisher:offer:sku:version) of the image from which to create a disk')
         c.argument('image_reference_lun', type=int, help='If the disk is created from an image\'s data disk, this is an index that indicates which of the data disks in the image to use. For OS disks, this field is null')
-        c.argument('gallery_image_reference', help='ID of the Compute Gallery image version from which to create a disk')
+        c.argument('gallery_image_reference', help='ID of the Compute, Shared or Community Gallery image version from which to create a disk')
+        c.ignore('gallery_image_reference_backend')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:185]: Would the naming be more accurate to change the `backend` to `type`?
```suggestion
        c.ignore('gallery_image_reference_type')
```

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 2]
```diff
@@ -181,7 +181,8 @@ def load_arguments(self, _):
         c.argument('disk_mbps_read_only', type=int, help='The total throughput (MBps) that will be allowed across all VMs mounting the shared disk as ReadOnly. MBps means millions of bytes per second - MB here uses the ISO notation, of powers of 10')
         c.argument('image_reference', help='ID or URN (publisher:offer:sku:version) of the image from which to create a disk')
         c.argument('image_reference_lun', type=int, help='If the disk is created from an image\'s data disk, this is an index that indicates which of the data disks in the image to use. For OS disks, this field is null')
-        c.argument('gallery_image_reference', help='ID of the Compute Gallery image version from which to create a disk')
+        c.argument('gallery_image_reference', help='ID of the Compute, Shared or Community Gallery image version from which to create a disk')
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:184]:
```suggestion
        c.argument('gallery_image_reference', help='ID of the Compute, Shared or Community Gallery image version from which to create a disk. For details about valid format, please refer to the help sample')
```

File: src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py [hunk 3]
```diff
@@ -1005,6 +1005,102 @@ def test_vm_disk_max_shares_etc(self, resource_group):
...
+                 '--os-type linux --os-state Specialized -p publisher1 -f offer1 -s sku1')
+        self.cmd('sig image-version create -g {rg} --gallery-name {gallery1} --gallery-image-definition {image1} '
+                 '--gallery-image-version {version} --virtual-machine {vm_id}')
+
+        public_name = self.cmd('sig show --gallery-name {gallery1} --resource-group {rg} --select Permissions', checks=[
+            self.check('provisioningState', 'Succeeded'),
+            self.check('length(sharingProfile.communityGalleryInfo.publicNames)', '1')
+        ]).get_output_in_json()['sharingProfile']['communityGalleryInfo']['publicNames'][0]
+        community_gallery_image_version = '/CommunityGalleries/' + public_name + '/Images/' + self.kwargs['image1'] \
+                                          + '/Versions/' + self.kwargs['version']
```
Comment (zhoxing-ms) [inline src/azure-cli/azure/cli/command_modules/vm/tests/latest/test_vm_commands.py:1044]: After feature #21423 is completed, `community_gallery_image_version` can be obtained from the `sig image-version show-community` command instead

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**